packages/shared/components/inputs/NetworkInput.svelte

```diff
@@ -1,48 +1,49 @@
 <script lang="ts">
     import { Modal, SelectorInput, IOption } from 'shared/components'
     import { activeProfile, getNetworkHrp } from '@core/profile'
-    import { NETWORK_ADDRESS, DestinationNetwork } from '@core/layer-2'
     import { validateBech32Address } from '@core/utils'
+    import { isIscpChain } from '@core/network'
+    import type { ChainConfiguration } from '@core/network'
 
-    const readonlyAttribute = $activeProfile?.isDeveloperProfile ? {} : { readonly: false }
-    const networkAddresses = NETWORK_ADDRESS[$activeProfile?.network?.id]
+    export let layer2ChainAddress: string | undefined = undefined
+    export let showLayer2: boolean = false
 
-    const LAYER_1_NETWORK_OPTION = {
-        key: DestinationNetwork.Shimmer,
-        value: networkAddresses[DestinationNetwork.Shimmer],
+    const readonlyAttribute = $activeProfile?.isDeveloperProfile ? {} : { readonly: true }
+    const layer1Network = {
+        key: $activeProfile?.network.name,
+        value: undefined,
     }
 
-    export let networkAddress: string = LAYER_1_NETWORK_OPTION.value
-    export let showLayer2: boolean = false
-
-    $: networkOptions = showLayer2 ? getLayer2NetworkOptions() : [LAYER_1_NETWORK_OPTION]
+    let inputElement: HTMLInputElement
+    let modal: Modal
+    let error: string
 
-    let selected: IOption =
-        getLayer2NetworkOptions().find((option) => option.value === networkAddress) ?? LAYER_1_NETWORK_OPTION
-    $: if (!showLayer2) {
-        selected = LAYER_1_NETWORK_OPTION
-    }
+    $: networkOptions = getNetworkOptions(showLayer2)
 
-    let inputElement: HTMLInputElement = undefined
-    let modal: Modal = undefined
+    let selected: IOption = networkOptions?.find((option) => option.value === layer2ChainAddress) ?? layer1Network
 
-    let error: string
+    $: layer2ChainAddress = selected?.value
 
-    $: networkAddress = selected?.value
+    function getNetworkOptions(showLayer2: boolean): IOption[] {
+        let layer2Networks: IOption[] = []
+        if (showLayer2) {
+            layer2Networks =
+                $activeProfile.network?.chainConfigurations?.map((chain) => ({
+                    key: chain.name,
+                    value: getNetworkValue(chain),
+                })) ?? []
+        }
```

Comment:
should we show chains in the dropdown if the value is `undefined` ?

Reply:
Yeas, because the value defines the alias address, but that's only available for ISCP chains.

Reply:
Not at the moment. Once we have EVM Chains, we need a better way of selecting the chain, maybe by introducing the `id` for the different options.